lib/Backend/Commands.hs

```diff
@@ -0,0 +1,453 @@
+module Backend.Commands where
+
+import Fmt ( pretty )
+import Control.Lens hiding (view)
+import CLI.Types
+import Polysemy
+import Control.Monad.State
+import Data.Time (getCurrentTime, UTCTime, utctDay)
+import qualified Data.Text as T
+import qualified Data.HashMap.Strict as HashMap
+import GHC.Exts (Down(..), sortWith)
+import Data.Maybe (fromMaybe)
+import Data.Time.Calendar.Compat (pattern YearMonthDay)
+import Data.Time.Calendar.Month.Compat (pattern MonthDay)
+import Control.Lens (view)
+import Data.Functor (($>))
+import qualified Data.List as List
+import Polysemy.Error (throw, Error, runError)
+import Control.Monad.Extra (whenM)
+
+import Backend (BackendEffect, listSecrets, readSecret, writeSecret, deleteSecret)
+import Entry (Entry, Field, FieldKey, value, fields, dateModified, newEntry, newField, visibility, FieldVisibility(..), EntryTag)
+import qualified Entry as E
+import Coffer.Directory (Directory)
+import qualified Coffer.Directory as Dir
+import Coffer.Path (Path, mkPath, mkPathSegment, unEntryPath, unPathSegment, entryPathName, EntryPath)
+import qualified Coffer.Path as Path
+import Error (CofferError(..))
+
+runCommand
+  :: (Member BackendEffect r, Member (Embed IO) r, Member (Error CofferError) r)
+  => Command res -> Sem r res
+runCommand = \case
+  CmdView opts -> catchAndReturn $ viewCmd opts
+  CmdCreate opts -> catchAndReturn $ createCmd opts
+  CmdSetField opts -> catchAndReturn $ setFieldCmd opts
+  CmdDeleteField opts -> deleteFieldCmd opts
+  CmdFind opts -> findCmd opts
+  CmdRename opts -> renameCmd opts
+  CmdCopy opts -> catchAndReturn $ copyCmd opts
+  CmdDelete opts -> catchAndReturn $ deleteCmd opts
+  CmdTag opts -> catchAndReturn $ tagCmd opts
+  where
+    catchAndReturn :: forall e r. Sem (Error e ': r) e -> Sem r e
+    catchAndReturn action =
+      either id id <$> runError @e action
```

Comment:
This function could be in some `Util` module.

Reply:
Agreed, I've added a `Util` module